src/odb/include/odb/db.h

```diff
@@ -9286,6 +9292,35 @@ class dbAccessPoint : public dbObject
   // User Code End dbAccessPoint
 };
 
+class dbGlobalConnect : public dbObject
+{
+ public:
+  // User Code Begin dbGlobalConnectEnums
+  // User Code End dbGlobalConnectEnums
+  dbRegion* getRegion() const;
+
+  dbNet* getNet() const;
+
+  std::string getInstPattern() const;
+
+  std::string getPinPattern() const;
+
+  // User Code Begin dbGlobalConnect
+  std::map<dbMaster*, std::set<dbMTerm*>> getMTerms() const;
+  std::set<dbInst*> getInsts() const;
+
+  bool appliesTo(dbInst* inst) const;
+  bool appliesTo(dbMTerm* mterm) const;
```

Comment:
I'm wondering it if makes sense to move more of the global connect functionality into odb and present a higher level API rather than these rather low level ones.  What do you think?

Reply:
I think it lives in PDN for legacy reasons, but probably shouldn't since any tool that makes an instance might need access to this. I think it would make sense to maybe add a function to the `dbBlock` to run the global connect rules. I'm happy to migrate it over to there along with associated tcl and documentation if you would like.

Reply:
Please do.  Thanks

Reply:
Do you think it makes sense for dbInst::create to automatically run global connect on itself?

Reply:
I would love that, but I think the regex might be too slow? maybe?
I think the mterms could be stored as dbId's and updated when the dbLibs are changed, so that might help speed that up, but the name regex there might not be much that can be done to speed that up. The resizer creates a lot of instances that are then destroyed if they didn't improve things, so running regex on all of those would be a waste?

Reply:
That's a fair point, it might be too much of a hit during optimization.